ftlib/commlib/pytorch/impl.py

```diff
@@ -51,6 +54,7 @@ def allreduce(self, data, op="MEAN"):
 
     @BasicCommLib.register_api
     def broadcast(self, data, root_rank):
+        data = torch.from_numpy(data) if isinstance(data, np.ndarray) else data
         dist.broadcast(data, root_rank)
```

Comment:
What about the return value though? The data type is also pytorch tensor here?

Reply:
I see. The broadcast is in-place but ElasticDL refers the API to return a numpy ndarray.

Reply:
Let's see the following example:

```
a = numpy.array([1,2,3]).astype(np.float32)

commlib.broadcast(data=a, root_rank=root_rank)
```

Since `torch.from_numpy` is used here, the underlying memory location is shared: https://pytorch.org/tutorials/beginner/blitz/tensor_tutorial.html#numpy-bridge

While the `data` within `broadcast` remains as a `torch.Tensor`, `a` is still a `numpy.ndarray`, of which the value is changed after `broadcast`.